test/integration_test.py

```diff
@@ -326,14 +335,27 @@ def _prune_test_bundles(self, bundle_fqids: List[BundleFQID], max_bundles: int)
                 break
         return filtered_bundle_fqids
 
+    def _filter_superseded_bundle_versions(self, bundle_fqids: Set[BundleFQID]) -> Set[BundleFQID]:
+        fqid_holder = {
+            uuid_: [fqid.version for fqid in items]
+            for uuid_, items in groupby(sorted(bundle_fqids, key=attrgetter("uuid", "version")),
+                                        key=attrgetter("uuid"))
+        }
+        return set(BundleFQID(k, v[-1]) for k, v in fqid_holder.items())
+
     @property
     def expected_fqids(self) -> KeysView[BundleFQID]:
         return self.notifications.keys()
 
     def _assert_bundles_are_indexed(self, entity_type: str):
         timeout = 600
         indexed_fqids = set()
-        expected_fqids = self.expected_fqids
+        expected_fqids = self._filter_superseded_bundle_versions(set(self.expected_fqids))
+        num_expected_bundles = len(expected_fqids)
+        superseded_bundles = self.expected_fqids - expected_fqids
+        if superseded_bundles:
+            log.debug('Inspections for bundles: %r were removed due to indexing a superseded version of the bundle',
+                      superseded_bundles)
```

Comment:
How do you feel about moving lines 355-358 into `_filter_superseded_bundle_versions()`? It'd help keep `_assert_bundles_are_indexed()` focused on it's purpose.

Reply:
i'm up for that modification.